featuretools/tests/synthesis/test_deep_feature_synthesis.py

```diff
@@ -1409,3 +1407,85 @@ def generate_name(self, base_feature_names):
+
+    features1 = ft.dfs(entityset=es,
+                       target_entity="customers",
+                       trans_primitives=trans_prims,
+                       groupby_trans_primitives=groupby_trans_prim,
+                       agg_primitives=agg_prims,
+                       where_primitives=where_prims,
+                       seed_features=seed_features,
+                       max_features=-1,
+                       max_depth=2,
+                       features_only=2)
+
+    trans_prims.reverse()
+    groupby_trans_prim.reverse()
+    agg_prims.reverse()
+    where_prims.reverse()
+    seed_features.reverse()
+
+    features2 = ft.dfs(entityset=es,
+                       target_entity="customers",
+                       trans_primitives=trans_prims,
+                       groupby_trans_primitives=groupby_trans_prim,
+                       agg_primitives=agg_prims,
+                       where_primitives=where_prims,
+                       seed_features=seed_features,
+                       max_features=-1,
+                       max_depth=2,
+                       features_only=2)
+
+    assert len(features1) == len(features2)
+
+    for i in range(len(features2)):
+        assert features1[i].unique_name() == features2[i].unique_name()
+
+
+def test_no_transform_stacking():
+    df1 = pd.DataFrame({"id": [0, 1, 2, 3],
+                        "A": [0, 1, 2, 3]})
+    df2 = pd.DataFrame({'first_id': [0, 1, 1, 3], 'B': [99, 88, 77, 66]})
+
+    entities = {"first": (df1, 'id'),
+                "second": (df2, 'index')}
+    relationships = [("first", 'id', 'second', 'first_id')]
+    es = ft.EntitySet("data", entities, relationships)
+
+    feature_defs = ft.dfs(entityset=es, target_entity='second',
+                          trans_primitives=['negate', 'add_numeric'],
+                          agg_primitives=['sum'],
+                          max_depth=4,
+                          features_only=2)
+    assert len(feature_defs) == 15
+
+    assert feature_with_name(feature_defs, 'first_id')
+    assert feature_with_name(feature_defs, 'B')
+    assert feature_with_name(feature_defs, '-(B)')
+    assert feature_with_name(feature_defs, 'first.A')
+    assert feature_with_name(feature_defs, 'first.SUM(second.B)')
+    assert feature_with_name(feature_defs, 'first.-(A)')
+    assert feature_with_name(feature_defs, 'B + first.A')
+    assert feature_with_name(feature_defs, 'first.SUM(second.-(B))')
+    assert feature_with_name(feature_defs, 'first.A + SUM(second.B)')
+    assert feature_with_name(feature_defs, 'first.-(SUM(second.B))')
+    assert feature_with_name(feature_defs, 'B + first.SUM(second.B)')
+    assert feature_with_name(feature_defs, 'first.A + SUM(second.-(B))')
+    assert feature_with_name(feature_defs, 'first.SUM(second.-(B)) + SUM(second.B)')
+    assert feature_with_name(feature_defs, 'first.-(SUM(second.-(B)))')
+    assert feature_with_name(feature_defs, 'B + first.SUM(second.-(B))')
```

Comment:
to save space, we could make a list of the expected feature names, compare this new list's length to `feature_defs` length, and loop over the expected names to check they are in `feature_defs`